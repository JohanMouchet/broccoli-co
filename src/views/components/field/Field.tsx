import React from "react";
import styles from "./Field.module.scss";

type FieldProps = {
  label?: string;
  type?: "text" | "email";
  name: string | undefined;
  children?: React.ReactNode;
  attributes?: Record<string, unknown>;
};

const Field: React.FC<FieldProps> = ({
  label,
  type = "text",
  name,
  children,
  attributes,
}) => (
  <label className={styles["label"]}>
    {label}
    <input
      type={type}
      name={name}
      className={styles["field"]}
      {...attributes}
    />
    {children && <span className={styles["validation"]}>{children}</span>}
  </label>
);

export default Field;
