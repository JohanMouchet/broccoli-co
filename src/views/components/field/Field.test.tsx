import React from "react";
import { render } from "@testing-library/react";
import Field from "./Field";

it("Renders <Field /> without crashing", () => {
  expect(
    render(<Field type="text" name="name" value="Value" />).container.firstChild
  ).toMatchSnapshot();
});
