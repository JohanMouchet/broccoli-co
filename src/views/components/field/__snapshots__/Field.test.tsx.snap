// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Field /> without crashing 1`] = `
<label
  class="label"
>
  <input
    class="field"
    name="name"
    type="text"
  />
</label>
`;
